temporian/implementation/numpy/operators/assign.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation for the Assign operator."""
+from typing import Dict, List
+
+import numpy as np
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+
+
+def _get_equal_sampling_indexes(
+    sampling_1: "NumpySampling", sampling_2: "NumpySampling", index: tuple
+) -> np.ndarray:
+    """Get indexes of timestamps that are equal to sampling_1 & sampling_2
+      in a specific sampling index.
+
+    Args:
+        sampling_1 (NumpySampling): sampling of assignee event.
+        sampling_2 (NumpySampling): sampling of assigned event.
+        index (tuple): index to check.
+
+    Returns:
+        np.ndarray: indexes of timestamps that are equal to sampling_1 & sampling_2.
+    """
+    return np.where(sampling_1.data[index] == sampling_2.data[index])[0]
```

Comment:
If I understand correctly, this is checking that for each position in the timestamps list, the timestamp is equal between both arrays. What happens if the two samplings have timestamps in common, but not in the exact same positions? E.g.:
```
s1.data[index] = [1, 2, 3]
s2.data[index] = [2, 3, 4]
```
in this case we would want the first and second positions in `s2` to be assigned to the second and third positions in `s1`. Can we write a test for this case?

Reply:
good catch, fixing it with extra border cases